frontend/lib/src/components/widgets/DataFrame/DataFrame.tsx

```diff
@@ -335,10 +352,59 @@ function DataFrame({
     }
   }, [element.formId, resetEditingState, widgetMgr])
 
+  const freezeColumns = isEmptyTable
+    ? 0
+    : columns.filter((col: BaseColumn) => col.isIndex).length
+
+  // Determine if the table requires horizontal or vertical scrolling:
+  React.useEffect(() => {
+    if (resizableContainerRef.current && !isEmptyTable) {
+      // TODO(lukasmasuch): This is only a hacky and temporary solution until
+      // glide-data-grid provides a better way to determine this:
+      // https://github.com/glideapps/glide-data-grid/issues/784
+
+      // Get the bounds of the data grid scroll area:
+      const scrollAreaBounds = resizableContainerRef.current
+        ?.querySelector(".dvn-stack")
+        ?.getBoundingClientRect()
+      if (scrollAreaBounds) {
+        setHasVerticalScroll(
+          scrollAreaBounds.height > resizableContainerRef.current.clientHeight
+        )
+        setHasHorizontalScroll(
+          scrollAreaBounds.width > resizableContainerRef.current.clientWidth
+        )
+      }
+    }
+  }, [resizableSize, numRows, isEmptyTable, glideColumns])
+
   return (
     <StyledResizableContainer
       data-testid="stDataFrame"
       className="stDataFrame"
+      ref={resizableContainerRef}
+      onMouseDown={e => {
+        if (resizableContainerRef.current && hasCustomizedScrollbars) {
+          // Prevent clicks on the scrollbar handle to propagate to the grid:
+          const boundingClient =
+            resizableContainerRef.current.getBoundingClientRect()
+
+          if (
+            hasHorizontalScroll &&
+            boundingClient.height - (WEBKIT_SCROLLBAR_SIZE + 1) <
```

Comment:
This +1 seems a little odd. I assume you're trying to make some more space for the click which makes sense. If so, I feel like a comment would be useful.

Reply:
Yep, indeed. Somehow I could still use the scrollbar and trigger unwanted selections even when it was one pixel out .. this is to prevent it. 
I will add a comment